### PR TITLE
Fixing problem interpreting default and computed values

### PR DESCRIPTION
**Pull request by lajones**
Fix for #7918.

Stop attempting to interpret the default/computed values stored by SQL Server. Just report them exactly as they are stored (including any parentheses).

Note: new entries needed in provider design-time services (@ajcvickers please see the changes I had to put into `SqlServerDesignTimeServices` and `SqliteDesignTimeServices`. I think these were missed as part of #8316).

### CODE REVIEW

File: src/EFCore.SqlServer.Design/Infrastructure/SqlServerDesignEventId.cs [hunk 1]
```diff
@@ -61,11 +61,13 @@ public enum SqlServerDesignEventId
 
         /// <summary>
         ///     Cannot interpret default value.
+        ///     Note: no longer used.
```
Comment (lajones) [inline src/EFCore.SqlServer.Design/Infrastructure/SqlServerDesignEventId.cs]: Did not want to delete this and the below as they are part of the public API.

File: src/EFCore.Sqlite.Design/Internal/SqliteDesignTimeServices.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.EntityFrameworkCore.Internal;
```
Comment (lajones) [inline src/EFCore.Sqlite.Design/Internal/SqliteDesignTimeServices.cs]: I will put this in correct alphabetical order.

File: src/EFCore.SqlServer.Design/Infrastructure/SqlServerDesignEventId.cs (hunk 1 above)
Comment (ajcvickers) [inline src/EFCore.SqlServer.Design/Infrastructure/SqlServerDesignEventId.cs]: You can delete them or not--all this is changing for 2.0 anyway. Fewer merge conflicts for me if you don't delete them now, and I will do it as part of my change.

File: src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs [hunk 3]
```diff
@@ -22,6 +23,8 @@ public class SqlServerDesignTimeServices : IDesignTimeServices
         /// </summary>
         public virtual void ConfigureDesignTimeServices(IServiceCollection serviceCollection)
             => serviceCollection
+                .AddSingleton<ILoggingOptions, LoggingOptions>()
+                .AddSingleton<IInterceptingLogger<LoggerCategory.Scaffolding>, InterceptingLogger<LoggerCategory.Scaffolding>>()
```
Comment (ajcvickers) [inline src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs]: Does this not work if registered as the open generic?

File: src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs [hunk 4]
```diff
@@ -22,6 +23,8 @@ public class SqlServerDesignTimeServices : IDesignTimeServices
         /// </summary>
         public virtual void ConfigureDesignTimeServices(IServiceCollection serviceCollection)
```
Comment (ajcvickers) [inline src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs]: @bricelam From what I can tell, in DesignTimeServicesBuilder, ConfigureProviderServices, which indirectly calls this, is called after ConfigureContextServices. This means that any service that has been obtained from the context will then be overwritten if GetProviderDesignTimeServices also tries to add it. So, for example, in this case ILoggingOptions from the context will be overwritten.

Is this intentional? Would it be better for these services to be TryAdded?

Comment (lajones) [inline src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs]: @ajcvickers @bricelam Probably still need the answer to the above, but note: Scaffolding does not call `ConfigureContextServices()`. That only happens if you call `DesignTimeServicesBuilder.Build(DbContext)`. Scaffolding does not have an existing `DbContext` so it calls `DesignTimeServicesBuilder.Build(string)`.

Comment (bricelam) [inline src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs]: The order is as follows.
* ConfigureServices
* ConfigureContextServices (if using a context)
* ConfigureProviderServices
* ConfigureUserServices

Given that, I think `ILoggingOptions` and `IInterceptingLogger<>` need to be added to `DesignTimeServicesBuilder.ConfigureServices` (and overridden by `ConfigureContextServices` when possible).

The design is intentional. If a provider needs to replace a default service at design-time, they can. Likewise if a user needs to override a provider service, they can.

Comment (lajones) [inline src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs]: @bricelam Would it be better to add them to [AddScaffolding()](https://github.com/aspnet/EntityFramework/blob/dev/src/EFCore.Design/Scaffolding/Internal/ScaffoldingServiceCollectionExtensions.cs#L23)?

Also how does the call to [AddLogging()](https://github.com/aspnet/EntityFramework/blob/dev/src/EFCore.Design/Design/Internal/DesignTimeServicesBuilder.cs#L80) in `DesignTimeServicesBuilder.ConfigureServices()` interact with this?

Comment (ajcvickers) [inline src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs]: @bricelam Thanks, that makes sense. @lajones Yes, they should be added to AddScaffolding. With regard to AddLogging, it adds external logging dependencies. These are our own logging abstractions.

File: src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs (hunk 3 above)
Comment (lajones) [inline src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs]: I was able to get it to work with the generic instead. So I did that.

File: src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs (hunk 4 above)
Comment (lajones) [inline src/EFCore.SqlServer.Design/Internal/SqlServerDesignTimeServices.cs]: Have checked in new commit implementing the above. Could one of you sign off please?

**Comment by lajones** [comment]
Checked in with commit 1d86186cd.

**Pull request closed without merging**